xbmc/cores/VideoPlayer/VideoPlayer.cpp

```diff
@@ -4494,23 +4494,42 @@ int CVideoPlayer::AddSubtitleFile(const std::string& filename, const std::string
 {
   std::string ext = URIUtils::GetExtension(filename);
   std::string vobsubfile = subfilename;
-  if (ext == ".idx")
+  if (ext == ".idx" || ext == ".sup")
   {
-    if (vobsubfile.empty()) {
-      // find corresponding .sub (e.g. in case of manually selected .idx sub)
-      vobsubfile = CUtil::GetVobSubSubFromIdx(filename);
+    std::shared_ptr<CDVDDemux> pDemux;
+    if (ext == ".idx")
+    {
       if (vobsubfile.empty())
+      {
+        // find corresponding .sub (e.g. in case of manually selected .idx sub)
+        vobsubfile = CUtil::GetVobSubSubFromIdx(filename);
+        if (vobsubfile.empty())
+          return -1;
+      }
+      auto pDemuxVobsub = std::make_shared<CDVDDemuxVobsub>();
```

Comment:
Some whitespace would help the readability of this section. I would add newlines after braces or return statements where applicable.

Reply:
thanks, I'll add some  blank lines.